spec/acceptance/zabbix_proxy_spec.rb

```diff
@@ -2,45 +2,49 @@
 require 'serverspec_type_zabbixapi'
 
 # rubocop:disable RSpec/LetBeforeExamples
-describe 'zabbix_proxy type', unless: default[:platform] =~ %r{debian-10-amd64} do
-  context 'create zabbix_proxy resources' do
-    # This will deploy a running Zabbix setup (server, web, db) which we can
-    # use for custom type tests
-    pp1 = <<-EOS
+describe 'zabbix_proxy type', unless: default[:platform] =~ %r{(ubuntu-16.04|debian-9|debian-10)-amd64} do
+  %w[4.0 5.0 5.2].each do |zabbix_version|
+    # 5.2 server packages are not available for RHEL 7
+    next if zabbix_version == '5.2' and default[:platform] == 'el-7-x86_64'
+    context "create zabbix_proxy resources with zabbix version #{zabbix_version}" do
+      # This will deploy a running Zabbix setup (server, web, db) which we can
+      # use for custom type tests
+      pp1 = <<-EOS
       class { 'apache':
         mpm_module => 'prefork',
       }
       include apache::mod::php
       class { 'postgresql::globals':
-        encoding => 'UTF-8',
         locale   => 'en_US.UTF-8',
         manage_package_repo => true,
         version => '12',
       }
       -> class { 'postgresql::server': }
 
       class { 'zabbix':
-        zabbix_version   => '4.4',
+        zabbix_version   => "#{zabbix_version}",
         zabbix_url       => 'localhost',
         zabbix_api_user  => 'Admin',
         zabbix_api_pass  => 'zabbix',
         apache_use_ssl   => false,
         manage_resources => true,
         require          => [ Class['postgresql::server'], Class['apache'], ],
       }
-    EOS
+      EOS
 
-    # Cleanup old database
-    prepare_host
+      # setup zabbix. Apache module isn't idempotent and requires a second run
+      it 'works with no error on first apply' do
+        # Cleanup old database
+        prepare_host
 
-    it 'works idempotently with no errors' do
-      # Run it twice and test for idempotency
-      apply_manifest(pp1, catch_failures: true)
-      apply_manifest(pp1, catch_changes: true)
-    end
+        apply_manifest(pp1, catch_failures: true)
```

Comment:
I don't know why though it was idempotent before. 
It's weird because we are running the same manifests on the other specs (application, host etc).
Apache module isn't idempotent.

Reply:
apache being apache ¯\_(ツ)_/¯. I think we can ignore this since it was idempotent before as well.